packages/gateway-client/src/worker/p2p-fetch/index.ts

```diff
@@ -13,9 +13,13 @@ const p2pFetch = async (
     const request = new P2pFetchRequest(p2pClient, givenReqObj, givenReqInit);
```

Comment:
First, check if the request object is not a request, if it isn't, pass all args to nativeFetch.

Reply:
Then, it is a request object, but do the local mode bootstrap url check.

Reply:
Fainlly, then constrcut and execute the request.